src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K,V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> targetNode = getNode(key);
+        if (targetNode != null) {
+            targetNode.value = value;
+            return;
+        }
+        if (size == threshold) {
+            resize();
+        }
+        addNode(getHash(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> targetNode = getNode(key);
+        return targetNode == null ? null : targetNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode());
+    }
+
+    private Node<K,V> getNode(K key) {
+        int hash = getHash(key);
+        Node<K, V> node = table[hash % table.length];
+        while (node != null) {
+            if (node.hash == hash && (node.key == key
+                    || (node.key != null && node.key.equals(key)))) {
+                return node;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    private void addNode(int hash, K key, V value) {
+        int index = hash % table.length;
```

Comment:
```suggestion
        int index = getIndex(hash);
```
lets extract it to a separate method